lib/zclient.c

```diff
@@ -1622,6 +1622,48 @@ static bool zapi_decode_prefix(struct stream *s, struct prefix *p)
 	return false;
 }
 
+static void zapi_encode_sockunion(struct stream *s, const union sockunion *su)
+{
+	int family = sockunion_family(su);
+	size_t addrlen = family2addrsize(family);
+
+	stream_putc(s, (uint8_t)family);
+
+	/*
+	 * degenerate case if unknown length
+	 */
+	if (!addrlen)
+		return;
+
+	stream_write(s, sockunion_get_addr(su), addrlen);
+}
+
+static bool zapi_decode_sockunion(struct stream *s, union sockunion *su)
+{
+	uint8_t family;
+	size_t addrlen;
+	uint8_t buf[sizeof(union sockunion)];
+
+	memset(su, 0, sizeof(*su));
+
+	STREAM_GETC(s, family);
+	sockunion_family(su) = family;
+
+	addrlen = family2addrsize(family);
+	if (!addrlen)
```

Comment:
Also here, can't we check earlier before doing other stuff?

Reply:
This one is in the receive path - can't detect error before family byte is read.